File: bubbletea.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int N; scanf("%d",&N);
    vector<int> tea_prices(N);
    for (int i = 0; i < N; i++) scanf("%d", &tea_prices[i]);
    int M; scanf("%d",&M);
    vector<int> top_prices(M);
    for (int i = 0; i < M; i++) scanf("%d", &top_prices[i]);
    int lowest = 1000000000;
    for (int i = 0; i < N; i++) {
        int K; scanf("%d",&K);
        for (int j = 0; j < K; j++) {
            int t; scanf("%d",&t);
            int curr = tea_prices[i] + top_prices[t-1];
            if (curr < lowest) lowest = curr;
        }
    }
    int X; scanf("%d",&X);
    printf("%d\n",max(X/lowest - 1,0));
    return 0;
}

File: bubbletea.py
from sys import stdin,stdout

N = int(stdin.readline())
tea_prices = list(map(int,stdin.readline().split()))
M = int(stdin.readline())
top_prices = list(map(int,stdin.readline().split()))
lowest = 10**9
for i in range(N):
    K,*toppings = map(lambda x: int(x) - 1,stdin.readline().split())
    for t in toppings:
        if tea_prices[i] + top_prices[t] < lowest:
            lowest = tea_prices[i] + top_prices[t]
X = int(stdin.readline())
stdout.write(f"{max(X//lowest - 1,0)}\n")
